### PR TITLE
chore: initial Vector Helm Chart and Helm Chart release process

### LINKED ISSUE #1397: Add Vector Kubernetes helm chart
We should add a Vector [helm chart](https://helm.sh/) to simplify Kubernetes installation.

### LINKED ISSUE #2674: K8s: prepare Helm charts
As part of #2222, we need to ship Helm charts for k8s. Once we have them implemented, we should also switch the dev/test flows to use Helm charts.

### LINKED ISSUE #2675: K8s: set up Helm chart release process
After #2674, as part of #2222 we should figure out how we ship Helm charts, and either set up a dedicated Helm repo, or configure snapshotting chart to S3, or prepare our github repo to be used as Helm repo.

### PR DESCRIPTION
**Pull request by MOZGIII**
~Will be rebased after #2653 is merged.~ :tada: Builds upon the work done at #2232.

- closes #2232
- closes #1397
- closes #2674 
- closes #2675

This PR adds the Helm Chart to deploy Vector into Kubernetes via Helm.

It has a lot of tweaks and knobs that we want to document.

- [x] add comments at `values.yaml`
- [x] check if `{}` should be `[]` at the default sources, transforms, and sinks

### CODE REVIEW

**Comment by Hoverbear** [comment]
Noting we seem to prefer `packages.timber.io` to `charts.timber.io`

**Comment by MOZGIII** [comment]
I meant the documentation on the website. It's quite a challenge to present and organize the docs all this.
I think I indeed should add more comments to the `values.yaml` - they can then be used as the source data for the docs on the website.

File: distribution/helm/vector/values.yaml [hunk 1]
```diff
@@ -39,53 +49,76 @@ securityContext: {}
...
+# Various tweakables for the `Pod`s managed by `DaemonSet`.
 resources: {}
 nodeSelector: {}
 tolerations: []
 affinity: {}
 
+# Additional volmes to pass to the `Pod`s managed by `DaemonSet`.
 extraVolumes: []
 
+# Additional volme mounts to pass to the `vector` container of the `Pod`s
```
Comment (ktff) [inline distribution/helm/vector/values.yaml:64]:
```suggestion
# Additional volume mounts to pass to the `vector` container of the `Pod`s
```

**Comment by MOZGIII** [comment]
This is merged, so the nightly builds should kick off, and we should see Helm Chart releases published to the repo.
If everything goes smoothly - the same CI job just needs to be copied to the release workflow.

**Comment by Xaeroxe** [comment]
Hi, my org is interested in running these helm charts. Where are they being published currently?

**Comment by binarylogic** [comment]
Hi @Xaeroxe, you can see some k8s installation examples [here](https://github.com/timberio/vector-k8s-examples). We are planning to release this officially in the next week or two. Let me know that helps.

**Comment by Xaeroxe** [comment]
It does, thank you!

**Pull request merged**